src/system/TLVPacketBufferBackingStore.h

```diff
@@ -87,25 +87,24 @@ class TLVPacketBufferBackingStore : public chip::TLV::TLVBackingStore
     bool mUseChainedBuffers;
 };
 
-class DLL_EXPORT PacketBufferTLVReader : public chip::TLV::TLVReader
+class DLL_EXPORT PacketBufferTLVReader : public TLV::ContiguousBufferTLVReader
 {
 public:
     /**
      * Initializes a TLVReader object to read from a PacketBuffer.
      *
-     * @param[in]    buffer  A handle to PacketBuffer, to be used as backing store for a TLV class.
-     * @param[in]    useChainedBuffers
-     *                       If true, advance to the next buffer in the chain once all data
-     *                       in the current buffer has been consumed.
+     * @param[in]    buffer  A handle to PacketBuffer, to be used as backing
+     *                       store for a TLV class.  If the buffer is chained,
```

Comment:
can we assert on this? I do not believe we have code that would assume just the head is used. When our code receives a 'packetbufferhandle' it eventually assumes that passing it on will  make all data handled.

Reply:
As in, assert that the buffer is not chained?  We have lots of code that knows it's working with non-chained buffers and either asserts or bails out if the buffers are chained....

I can add an assert here, I guess.  What's the "right" way to do that (presumably not `assert`)?

Reply:
@andy31415 After #9599 we will definitely never see chained buffers in any code that uses this class.